app/src/appcoinssdk/java/com/aptoide/iabexample/utilssdk/ApplicationUtils.java

```diff
@@ -0,0 +1,169 @@
+          "Result code was OK but in-app billing response was not OK: " + getResponseDesc(
+              responseCode), null, null);
+    } else if (resultCode == Activity.RESULT_CANCELED) {
+
+      logDebug("Purchase canceled - Response: " + getResponseDesc(responseCode));
+      purchaseFinishedListener.onPurchaseFinished(IABHELPER_USER_CANCELLED,
+          "Purchase canceled - Response: " + getResponseDesc(responseCode), null, null);
+    } else {
+      logError("Purchase failed. Result code: " + resultCode + ". Response: " + getResponseDesc(
+          responseCode));
+      purchaseFinishedListener.onPurchaseFinished(IABHELPER_UNKNOWN_PURCHASE_RESPONSE,
+          "Purchase canceled - Response: " + getResponseDesc(responseCode), null, null);
+    }
+    return true;
+  }
+
+  static boolean verifySignature(String signature, String purchaseData, String dataSignature) {
+    if (!Security.verifyPurchase(signature, purchaseData, dataSignature)) {
+      logError("Purchase signature verification FAILED");
+      return false;
+    }
+    return true;
+  }
+
+  static int getResponseCodeFromIntent(Intent i) {
+    Object o = i.getExtras()
+        .get(Utils.RESPONSE_CODE);
+    if (o == null) {
+      logError("Intent with no response code, assuming OK (known issue)");
+      return ResponseCode.OK.getValue();
+    } else if (o instanceof Integer) {
+      return ((Integer) o).intValue();
+    } else if (o instanceof Long) {
+      return (int) ((Long) o).longValue();
+    } else {
+      logError("Unexpected type for intent response code.");
+      logError(o.getClass()
+          .getName());
+      throw new RuntimeException("Unexpected type for intent response code: " + o.getClass()
+          .getName());
+    }
+  }
+
+  static void logDebug(String msg) {
+    Log.d(TAG, msg);
+  }
+
+  static void logError(String msg) {
+    Log.e(TAG, "In-app billing error: " + msg);
+  }
+
+  static String getTokenFromJSON(JSONObject data) {
+    return data.optString("purchaseToken");
+  }
+
+  static String getSkuFromJSON(JSONObject data) {
+    return data.optString("productId");
+  }
+
+  public static String getResponseDesc(int code) {
+    String[] iab_msgs = ("0:OK/1:User Canceled/2:Unknown/"
+        + "3:Billing Unavailable/4:Item unavailable/"
+        + "5:Developer Error/6:Error/7:Item Already Owned/"
+        + "8:Item not owned").split("/");
+    String[] iabhelper_msgs = ("0:OK/-1001:Remote exception during initialization/"
+        + "-1002:Bad response received/"
+        + "-1003:Purchase signature verification failed/"
+        + "-1004:Send intent failed/"
+        + "-1005:User cancelled/"
+        + "-1006:Unknown purchase response/"
+        + "-1007:Missing token/"
+        + "-1008:Unknown error/"
+        + "-1009:Subscriptions not available/"
+        + "-1010:Invalid consumption attempt").split("/");
+
+    if (code <= -1000) {
+      int index = -1000 - code;
+      if (index >= 0 && index < iabhelper_msgs.length) {
```

Comment:
the index >=0 condition is always true due to the if(code <= -1000) condition. so remove one or another